src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -378,14 +385,20 @@ public async Task<IActionResult> SetDeploymentTarget(string sessionId, [FromBody
                     return NotFound($"Recommendation {input.NewDeploymentRecipeId} used in existing deployment {existingDeployment.RecipeId} not found.");
                 }
 
-                var existingCloudApplicationMetadata = await templateMetadataReader.LoadCloudApplicationMetadata(input.ExistingDeploymentName);
-                state.SelectedRecommendation = state.SelectedRecommendation.ApplyPreviousSettings(existingCloudApplicationMetadata.Settings);
+                IDictionary<string, object> previousSettings;
+                if (existingDeployment.ResourceType == CloudApplicationResourceType.CloudFormationStack)
+                    previousSettings = (await templateMetadataReader.LoadCloudApplicationMetadata(existingDeployment.Name)).Settings;
+                else
+                    previousSettings = await deployedApplicationQueryer.GetPreviousSettings(existingDeployment);
```

Comment:
(minor, at your discretion) This function is getting long. To help keep it from growing, it looks like this "get previous settings" code could be placed in its own method to improve readability.